utils/configcpp/configcpp.cpp

```diff
@@ -95,46 +79,35 @@ void Config::checkAndReloadConfig()
   }
 }
 
+Config& Config::globConfigInstance()
+{
+  std::string configFilePath =
+      std::string(MCSSYSCONFDIR) + std::string("/columnstore/") + configDefaultFileName();
+  static Config config(configFilePath);
+  return config;
+}
+
 Config* Config::makeConfig(const string& cf)
 {
-  if (cf.empty() || cf == configDefaultFileName)
+  if (cf.empty() || cf == configDefaultFileName())
   {
-    if (!globHasConfig.load(std::memory_order_relaxed))
-    {
-      // To save against the moment zero race when multiple threads hits
-      // this scope.
-      boost::mutex::scoped_lock lk(fInstanceMapMutex);
-      if (globConfigInstancePtr)
-      {
-        globConfigInstancePtr->checkAndReloadConfig();
-        return globConfigInstancePtr.get();
-      }
-
-      // Make this configurable at least at compile-time.
-      std::string configFilePath =
-          std::string(MCSSYSCONFDIR) + std::string("/columnstore/") + configDefaultFileName;
-      globConfigInstancePtr.reset(new Config(configFilePath));
-      globHasConfig.store(true, std::memory_order_relaxed);
-      return globConfigInstancePtr.get();
-    }
-
-    boost::mutex::scoped_lock lk(fInstanceMapMutex);
-    globConfigInstancePtr->checkAndReloadConfig();
-    return globConfigInstancePtr.get();
+    boost::mutex::scoped_lock lk(instanceMapMutex());
```

Comment:
Are you sure about this replacement changing atomic with a mutex?

Reply:
Atomic was used for Singleton safety, C++11 makes Singleton safe itself